src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -23,8 +45,76 @@ public RewardScoreEntity recalculateScore(AllRewardScoresEntity allRewardScores,
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
         RewardScoreEntity fitnessScoreBefore = allRewardScores.getFitness();
 
-        // TODO calculate fitness score
+        double fitnessDecrease = calculateFitnessDecrease(contents, event);
+        double newFitnessScore = fitnessScoreBefore.getValue() * (1 - fitnessDecrease);
+        int newFitnessScoreInt = (int) Math.round(newFitnessScore);
+
+        // Decrease the fitness score
```

Comment:
why does fitness decrease when content was learned? That is not the correct logic. The fitness should increase! I think you have to look at the formulas in the wiki again.

Reply:
![image](https://github.com/IT-REX-Platform/reward_service/assets/34026908/d7b78329-2440-4fd8-b265-1709fbbe82a8)

Reply:
(there is a small error in the formula in the wiki, should be (100 - fitness) in the parenthesis)